tools/org.eclipse.n4js.hlc.base/src/org/eclipse/n4js/hlc/base/N4jscBase.java

```diff
@@ -407,6 +423,14 @@ public SuccessExitStatus doMain(String... args) throws ExitCodeException {
 			// help.
 			initInjection(refProperties());
 
+			// Register subgenerators
+			subgeneratorsRegistry.register(ecmaScriptSubGenerator.get(), N4JSGlobals.N4JS_FILE_EXTENSION);
+			subgeneratorsRegistry.register(ecmaScriptSubGenerator.get(), N4JSGlobals.N4JSX_FILE_EXTENSION);
+			subgeneratorsRegistry.register(ecmaScriptSubGenerator.get(), N4JSGlobals.JS_FILE_EXTENSION);
+
+			headless.registerComposedGenerator(n4jsCompositeGenerator);
+			headless.registerComposedGenerator(n4jsxCompositeGenerator);
+
```

Comment:
those registrations need to be inside platform check below!!

Reply:
Done. Moved into the check.